kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/PageToken.java

```diff
@@ -0,0 +1,151 @@
+package io.delta.kernel.internal.replay;
+
+import static java.util.Objects.requireNonNull;
+
+import io.delta.kernel.data.Row;
+import io.delta.kernel.internal.data.GenericRow;
+import io.delta.kernel.types.*;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+/** Page Token Class for Pagination Support */
+public class PageToken {
+  /** Variables to know where last page ends (current page starts) */
+  private final String startingFileName; // starting Log File Name (also last read log file name)
```

Comment:
does this apply to sidecars? only json and mulit-part-checkpoints / log compactions?

when we are reading a sidecar, what is this value?

Reply:
That's helpful, thanks! Can we update the variable documentation to explain this? e.g. how should this variable be used in relation to `sidecarIndex`.

And would `startingLogFileName` be a bit more accurate variable name?